packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:path/path.dart';
+
+import '../charts/treemap.dart';
+import 'vm_treemap.dart';
+
+class CodeSizeController {
+  ValueListenable<TreemapNode> get root => _root;
+  final _root = ValueNotifier<TreemapNode>(null);
+
+  Future<void> loadTree(String filename) async {
+    final codeSizePath = current + '/lib/src/code_size/';
+    final inputJson = File(codeSizePath + filename);
+
+    await inputJson.readAsString().then((inputJsonString) {
+      final inputJsonMap = json.decode(inputJsonString);
+
+      // Build a [Map] object containing heirarchical information for [inputJsonMap].
+      final processedJsonMap = treemapFromJson(inputJsonMap);
+      
+      // Set name for root node.
+      processedJsonMap['n'] = 'Root';
+
+      // Build a tree with [TreemapNode] from [processedJsonMap].
+      final newRoot = generateTreemapNode(processedJsonMap);
+
+      changeRoot(newRoot);
+    });
+  }
+
+  /// Builds a tree with TreemapNode from [treeJson] which represents
+  /// the hierarchical structure of the tree.
+  TreemapNode generateTreemapNode(Map<String, dynamic> treeJson) {
```

Comment:
rename to generateTree as it really is generating the entire tree, not a single node

Reply:
Done